PathPlanning/TimeBasedPathPlanning/SpaceTimeAStar.py

```diff
@@ -0,0 +1,221 @@
+"""
+Space-time A* Algorithm
+    This script demonstrates the Space-time A* algorithm for path planning in a grid world with moving obstacles.
+
+    Reference: https://www.davidsilver.uk/wp-content/uploads/2020/03/coop-path-AIWisdom.pdf
+"""
+
+import numpy as np
+import matplotlib.pyplot as plt
+from PathPlanning.TimeBasedPathPlanning.GridWithDynamicObstacles import (
+    Grid,
+    ObstacleArrangement,
+    Position,
+)
+import heapq
+from collections.abc import Generator
+import random
+from functools import total_ordering
+
+# Seed randomness for reproducibility
+RANDOM_SEED = 50
+random.seed(RANDOM_SEED)
+np.random.seed(RANDOM_SEED)
+
+@total_ordering # so the linter will chill about not implementing __gt__, __ge__, etc
+class Node:
```

Comment:
dataclass cannot be used?

Reply:
Changed to use a dataclass:
https://github.com/AtsushiSakai/PythonRobotics/pull/1170/commits/ba9100174d3e5cf34286a07e69fd3ac8479e7f81#diff-867b5aaee46b7be35687893a1fdacc8e11afa1777a4f60958cd5cdf6fa3d1f50R29